Rubberduck.Main/Root/TypeExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Rubberduck.Parsing.Common;
+using Rubberduck.Settings;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Rubberduck.Root
+{
+    internal static class TypeExtensions
+    {
+        internal static bool NotDisabledExperimental(this Type type, GeneralSettings initialSettings)
+        {
+            var attribute = type.CustomAttributes.FirstOrDefault(f => f.AttributeType == typeof(ExperimentalAttribute));
+            var ctorArg = attribute?.ConstructorArguments.Any() == true ? (string)attribute.ConstructorArguments.First().Value : string.Empty;
+
+            return attribute == null || initialSettings.EnableExperimentalFeatures.Any(a => a.Key == ctorArg && a.IsEnabled);
+        }
+
+        internal static bool IsBasedOn(this Type type, Type allegedBase)
```

Comment:
This is something a lot more restrictie then `BasedOn<type>` from CW. This code tests for direct concrete inheritance, but `BasedOn<type>` works for all implemented interfaces as well.

Reply:
Comment is outdated now.